packages/commerce-sdk-react/src/hooks/types.ts

```diff
@@ -70,3 +62,21 @@ export type DataType<T extends (arg: any) => Promise<unknown>> = T extends (
 ) => Promise<Response | infer R>
     ? R
     : never
+
+/**
+ * Modified version of React Query's Mutation Function. Added a second argument
+ * API clients.
+ */
+export type MutationFunction<TData = unknown, TVariables = unknown> = (
+    variables: TVariables,
+    apiClients: ApiClients
+) => Promise<TData>
+
+/**
+ * Modified version of React Query's Query Function. Added a second argument
+ * API clients.
+ */
+export type QueryFunction<TData = unknown, TQueryKey extends QueryKey = unknown[]> = (
+    context: QueryFunctionContext<TQueryKey>,
+    apiClients: ApiClients
+) => TData | Promise<TData>
```

Comment:
Are we expecting it could return TData here? I checked React-query [doc](https://tanstack.com/query/v4/docs/reference/useQuery), it looks like the QUeryFunction return Promise<TData> only